core/src/androidTest/java/org/hisp/dhis/android/core/dataset/internal/DataSetCompleteRegistrationPostCallMockIntegrationShould.kt

```diff
@@ -0,0 +1,139 @@
+            .bySyncState().eq(State.SYNCED)
+
+        assertEquals(toPostRepository.blockingGet().size, 1)
+        assertEquals(syncedRepository.blockingGet().size, 0)
+
+        // When the object is uploaded
+        d2.dataSetModule().dataSetCompleteRegistrations().blockingUpload()
+
+        // Then the status is synced
+        assertEquals(toPostRepository.blockingGet().size, 0)
+        assertEquals(syncedRepository.blockingGet().size, 1)
+    }
+
+    @Test
+    fun post_dataset_delete_complete_registration_success() {
+        val toPostRepository = d2.dataSetModule()
+            .dataSetCompleteRegistrations()
+            .bySyncState().eq(State.TO_POST)
+
+        val syncedRepository = d2.dataSetModule()
+            .dataSetCompleteRegistrations()
+            .bySyncState().eq(State.SYNCED)
+
+        val toUpdateRepository = d2.dataSetModule()
+            .dataSetCompleteRegistrations()
+            .bySyncState().eq(State.TO_UPDATE)
+
+        val dataSetCompleteRegistration = provideDataSetCompleteRegistration()
+        var dObject = d2.dataSetModule().dataSetCompleteRegistrations().value(
+            period = dataSetCompleteRegistration.period(),
+            organisationUnit = dataSetCompleteRegistration.organisationUnit(),
+            dataSet = dataSetCompleteRegistration.dataSet(),
+            attributeOptionCombo = dataSetCompleteRegistration.attributeOptionCombo()
+        )
+
+        dhis2MockServer.enqueueMockResponse(200, "imports/data_value_import_summary_web_response.json")
+        dObject.blockingSet()
+        assertEquals(toPostRepository.blockingGet().size, 1)
+        assertEquals(syncedRepository.blockingGet().size, 0)
+
+        d2.dataSetModule().dataSetCompleteRegistrations().blockingUpload()
+
+        assertEquals(toPostRepository.blockingGet().size, 0)
+        assertEquals(syncedRepository.blockingGet().size, 1)
+
+        dhis2MockServer.enqueueMockResponse(200)
+        dObject = d2.dataSetModule().dataSetCompleteRegistrations().value(
+            period = dataSetCompleteRegistration.period(),
+            organisationUnit = dataSetCompleteRegistration.organisationUnit(),
+            dataSet = dataSetCompleteRegistration.dataSet(),
+            attributeOptionCombo = dataSetCompleteRegistration.attributeOptionCombo()
+        )
+        dObject.blockingDelete()
+        assertEquals(toUpdateRepository.blockingGet().size, 1)
+
+        d2.dataSetModule().dataSetCompleteRegistrations().blockingUpload()
+
+        assertEquals(toUpdateRepository.blockingGet().size, 0)
+    }
+
+    private fun provideDataSetCompleteRegistration() = DataSetCompleteRegistration.builder()
+        .period("2018")
+        .dataSet("BfMAe6Itzgt")
+        .attributeOptionCombo("bRowv6yZOF2")
+        .organisationUnit("DiszpKrYNg8")
+        .date(dateFormat.parse("2010-03-02"))
+        .storedBy("android")
+        .build()
+
+    private val dateFormat = SimpleDateFormat("yyyy-MM-dd")
```

Comment:
Replace by `DateUtils.SIMPLE_DATE_FORMAT` to avoid adding Java dependencies. The implementation of SIMPLE_DATE_FORMAT still depends on Java, but it is centralized in a single class at least.